apis/spark/src/main/scala/TileDBVCFHailReader.scala

```diff
@@ -0,0 +1,263 @@
+        PCanonicalCall(false)
+      else {
+        vcfReader.infoAttributes.get(key).datatype match {
+          case AttributeDatatype.UINT8 => PInt32()
+          case AttributeDatatype.INT32 => PInt32()
+          case AttributeDatatype.CHAR => PCanonicalString()
+          case AttributeDatatype.FLOAT32 => PFloat32()
+        }
+      }
+    }
+    (infoAttr.toString.split("_")(1) -> dt.virtualType)
+  }
+
+  val vaSignature = PCanonicalStruct(Array(
+    PField("rsid", PCanonicalString(), 0),
+    PField("qual", PFloat64(), 1),
+    PField("filters", PCanonicalSet(PCanonicalString(true)), 2)), true)
+
+  val fullMatrixType: MatrixType = MatrixType(
+    globalType = TStruct.empty,
+    colType = TStruct("s" -> TString),
+    colKey = Array("s"),
+    rowType = TStruct(
+      "locus" -> TLocus(ReferenceGenome.GRCh37),
+      "alleles" -> TArray(TString)) ++ vaSignature.virtualType ++ (TStruct("info" -> TStruct(info:_*))),
+    rowKey = Array("locus", "alleles"),
+    entryType = entryType)
+
+  override def apply(tr: TableRead, ctx: ExecuteContext): TableValue = {
+    val requestedType = tr.typ
+    val rowType = tr.typ.rowType
+
+    val entryType: TStruct = rowType.fieldOption(LowerMatrixIR.entriesFieldName) match {
+      case Some(entriesArray) => entriesArray.typ.asInstanceOf[TArray].elementType.asInstanceOf[TStruct]
+      case None => TStruct.empty
+    }
+
+    if (df == null) {
+      df = ctx.backend.asSpark("spark")
+        .sparkSession.read
+        .format("io.tiledb.vcf")
+        .option("samples", samples.get)
+        .option("uri", uri)
+        .load()
+    }
+
+    //    var projection = df.groupBy("contig","posStart", "alleles")
+    //      .agg(functions.collect_list("genotype").as("GT"),
+    //        functions.collect_list("fmt_DP").as("DP"),
+    //        functions.collect_list("fmt_PL").as("PL"))
+
+    var projection = df
+      .withColumnRenamed("fmt_AD", "AD")
+      .withColumnRenamed("fmt_DP", "DP")
+      .withColumnRenamed("fmt_GQ", "GQ")
+      .withColumnRenamed("genotype", "GT")
+      .withColumnRenamed("fmt_MIN_DP", "MIN_DP")
+      .withColumnRenamed("fmt_PL", "PL")
+      .withColumnRenamed("fmt_SB", "SB")
+
+
+    var columns = Seq[String]("contig", "posStart", "alleles")
+
+    // Construct a map of info fields and row indices
+    var idx = 3
+    val infoFieldIdx: mutable.HashMap[String, Int] = mutable.HashMap()
+
+    // Check if the row has an "info" field
+    val hasInfo = rowType.hasField("info")
+    val infoFieldNames = info.map(_._1)
+    if (hasInfo) {
+      info.foreach { infoField =>
+        columns = columns :+ s"info_${infoField._1}"
+        infoFieldIdx.put(infoField._1, idx)
+        idx += 1
+      }
+    }
+
+    val hasRsid = rowType.hasField("rsid")
+    val hasQual = rowType.hasField("qual")
+    val hasFilter = rowType.hasField("filters")
+
+    // Parse info entry fields
+    val infoFields = Seq("AD", "DP", "GQ", "GT", "MIN_DP", "PL", "SB")
```

Comment:
There can be a lot more info fields than these, we should use the API we have to list info fields.